controllers/storagecluster/cephblockpools.go

```diff
@@ -257,12 +257,19 @@ func (o *ocsCephBlockPools) reconcileNonResilientCephBlockPool(r *StorageCluster
 			}
 			cephBlockPool.Spec.PoolSpec.EnableRBDStats = true
 
-			if storageCluster.Spec.Mirroring != nil {
-				if storageCluster.Spec.Mirroring.Enabled {
+			deploymentType, ok := storageCluster.Annotations["ocs.openshift.io/deployment-mode"]
+			if !ok {
+				return fmt.Errorf("deployment mode annotation not found")
+			}
+
+			// Since provider mode handles mirroring, we only need to handle for converged mode
+			if deploymentType != "provider" {
```

Comment:
```suggestion
			// Since provider mode handles mirroring, we only need to handle for internal mode
			if storageCluster.Annotations["ocs.openshift.io/deployment-mode"] != "provider" {
```

Reply:
Annotation is guaranteed to be there in all cases ?

Reply:
No, it will be present only in provider mode, but if we query an empty map with a key that is not present it will return an empty value